### PR TITLE
Add ability to share posts.

### PR DESCRIPTION
**Pull request by camporter**
For #536 adds the ability to share posts. Either the linked post URL or the post directly will be shared.

### CODE REVIEW

File: app/src/main/java/com/jerboa/Utils.kt [hunk 1]
```diff
@@ -374,6 +374,32 @@ fun LazyListState.isScrolledToEnd(): Boolean {
     return out
 }
 
+/**
+ * Open a sharesheet for the given post.
+ * If the post has a URL, share that.
+ * Otherwise share the post directly.
+ */
+fun sharePost(post: Post, ctx: Context) {
+    post.url?.let {
```
Comment (a1studmuffin) [inline app/src/main/java/com/jerboa/Utils.kt:383]: Perhaps we could break this out into two share actions so the user can choose "Share link" or "Share post"? That's how Boost and Bacon Reader work.

Comment (camporter) [reply app/src/main/java/com/jerboa/Utils.kt:383]: Sounds good. I wasn't sure what the feeling was on multiple share links.

Comment (camporter) [reply app/src/main/java/com/jerboa/Utils.kt]: Added both Share Link and Share Post.

File: app/src/main/java/com/jerboa/ui/components/post/PostListing.kt [hunk 2]
```diff
@@ -1366,6 +1384,18 @@ fun PostOptionsDialog(
...
                     },
                 )
+                postView.post.url?.also { url ->
+                    IconAndTextDrawerItem(
+                        text = stringResource(R.string.post_listing_share_link),
+                        icon = Icons.Outlined.Share,
+                        onClick = { onShareClick(url) },
+                    )
+                }
+                IconAndTextDrawerItem(
```
[the remark this answers is not in the document]
Comment (camporter) [reply app/src/main/java/com/jerboa/ui/components/post/PostListing.kt:1394]: Hi, this sharing of the `ap_id` (Share Post) should be outside of the check for the post url (Share Link) already. Let me know if I missed something, thanks.

**Review by twizmwazin — APPROVED** [review]

File: app/src/main/java/com/jerboa/ui/components/post/PostActivity.kt [hunk 3]
```diff
@@ -47,6 +47,8 @@ import com.jerboa.db.AccountViewModel
 import com.jerboa.getCommentParentId
 import com.jerboa.getDepthFromComment
 import com.jerboa.isModerator
+import com.jerboa.openLink
```
Comment (ZJouba) [inline app/src/main/java/com/jerboa/ui/components/post/PostActivity.kt:50]:
```suggestion
```

Comment (camporter) [reply app/src/main/java/com/jerboa/ui/components/post/PostActivity.kt]: Fixed.

**Comment by camporter** [comment]
Rebased again.

**Pull request merged**